quanda/benchmarks/config_parser.py

```diff
@@ -0,0 +1,435 @@
+from quanda.utils.training.options import optimizers, criteria, schedulers
+
+
+class BenchConfigParser:
+    """Parser for benchmark configurations."""
+
+    @classmethod
+    def load_metadata(
+        cls,
+        cfg: dict,
+        bench_save_dir: str = ".tmp",
+        load_meta_from_disk: bool = True,
+    ):
+        """Parse metadata configuration and return the metadata directory."""
+        meta_id = cfg.get("meta_id", f"{cfg['id']}_metadata")
+        repo_id = f"{cfg['repo_id']}/{meta_id}"
+        base_metadata_dir = os.path.join(bench_save_dir, "metadata")
+        # create metadata_dir if it doesn't exist
+        os.makedirs(base_metadata_dir, exist_ok=True)
+        metadata_dir = os.path.join(base_metadata_dir, f"{cfg['id']}_metadata")
+        if os.path.exists(metadata_dir) or not load_meta_from_disk:
+            return metadata_dir
+        return snapshot_download(
+            repo_id=repo_id, local_dir=metadata_dir, repo_type="dataset"
+        )
+
+    @classmethod
+    def parse_dataset_cfg(
+        cls,
+        ds_config: Optional[dict],
+        metadata_dir: str = ".tmp/meta",
+        bench_save_dir: str = ".tmp",
+        load_meta_from_disk: bool = True,
+    ):
+        """Return the dataset using the given parameters."""
+        if ds_config is None:
+            return None
+
+        dataset = cls._load_dataset_from_cfg(
+            ds_config, bench_save_dir, load_meta_from_disk
+        )
+
+        wrapper = copy.deepcopy(ds_config.get("wrapper", None))
+        if wrapper is not None:
+            return cls._apply_wrapper(
+                dataset, wrapper, metadata_dir, load_meta_from_disk
+            )
+        return dataset
+
+    @classmethod
+    def parse_model_cfg(
+        cls,
+        model_cfg: dict,
+        bench_save_dir: str,
+        repo_id: str,
+        cfg_id: str,
+        offline: bool,
+        device: str,
+    ) -> Tuple[torch.nn.Module, List[str], Callable]:
+        """Parse model configuration and return the model and checkpoints.
+
+        Parameters
+        ----------
+        model_cfg : dict
+            Model configuration dictionary
+        bench_save_dir : str
+            Path to checkpoint directory "ckpt".
+        repo_id : str
+            Repo ID Hugging Face
+        cfg_id : str
+            Configuration ID
+        offline : bool
+            If True, the method tries to load the model from the local cache.
+        device : str
+            Device to use for the model.
+
+        Returns
+        -------
+        Tuple[torch.nn.Module, List[str]]
+            The configured model and list of checkpoints
+
+        """
+        module_cfg = model_cfg["module"]
+        module_cls = pl_modules[module_cfg["name"]]
+        module = module_cls(**module_cfg["args"])
+
+        checkpoint_path = os.path.join(bench_save_dir, "ckpt")
+        ckpt_dir = cls.get_ckpt_folder(model_cfg, checkpoint_path, cfg_id)
+        ckpt_id = f"{repo_id}/{cfg_id}"
```

Comment:
@gumityolcu We have a problem: right now it is hardcoded that only a single ckpt is expected.

For the LDS metric, could you replace this with something like 
`ckpt_ids = [f"{repo_id}/{ckpt_id}" for ckpt_id in model_cfg.get("ckpt_ids", [])]`

Maybe ckpt_ids need to be even more structured, but the baseline would be something like:
ckpt ids:
   - lds_ckpt_subset_0
   - lds_ckpt_subset_1
   ...